Bump webpack from 5.91.0 to 5.94.0 in the npm_and_yarn group across 1 directory (#96)
rsksmart/devportal: rif/rns/architecture/rsk-registrar/registrars/fifs.md

- [Smart contract](https://github.com/rnsdomains/rns-rskregistrar/blob/master/contracts/FIFSRegistrar.sol)

Has registration role in `RSK Owner`.

- Defines a commit-reveal process to avoid front-running.
- Accepts payments via
  - ERC-20 `approve()` + `register()`.<sup>1</sup>
  - ERC-721 `transferAndCall()`.<sup>2</sup>
- Calculates price using [`NamePrice` contract](../other/nameprice).
- It has an owner that can<sup>3</sup>
  - Set minimum commitment age.
  - Set minimum registration name length available.
  - Change name price contract.

The registration must be performed as follows:

0. Calculate `makeCommitment` hash of the domain to be registered (off-chain).
1. Commit the calculated hash using `commit`.
2. Wait `minCommitmentAge` seconds.
3. Execute registration via ERC-20 (with approval) or ERC-677.

> Find `transferAndCall()` encoder in `utils/index.js`

## Methods

- [`price`](#price)
- [`makeCommitment`](#makecommitment)
- [`commit`](#commit)
- [`canReveal`](#canreveal)
- [`register`](#register)

### `price`

```solidity
function price (string memory name, uint expires, uint duration) public view returns(uint);
```

> Price of a name in RIF

Reviewer: ```suggestion
Calculates the price of a name, denominated in RIF.
```

- `name` not used. Pass `''`
- `expires` not used. Pass `0`
- `duration` to register the name for.

### `makeCommitment`

```
function makeCommitment (bytes32 label, address nameOwner, bytes32 secret) public pure returns (bytes32);
```

Create a commitment for register action.

> Don't use this method on-chain when committing.

Reviewer: ```suggestion
:warning:  Don't use this method on-chain, as that will reveal your `secret`.
Instead run this method off-chain (note that it is `pure`), and save the return value.
```

- `label` keccak256 of the name to be registered.
- `nameOwner` Owner of the name to be registered.
- `secret` Secret to protect the name to be registered.

> Returns The commitment hash.

Reviewer: ```suggestion
Returns the commitment hash.
```

### `commit`

```solidity
function commit(bytes32 commitment) external;
```

Commit before registering a name.

A valid commitment can be calculated using makeCommitment off-chain.

- `commitment` A valid commitment hash.

### `canReveal`

```solidity
function canReveal(bytes32 commitment) public view returns (bool);
```

Ensure the commitment is ready to be revealed.

This method can be polled to ensure registration.

- `commitment` Commitment to be queried.

> Returns Wether the commitment can be revealed or not.

Reviewer: ```suggestion
Returns whether the commitment can be revealed or not.
```

### `register`

```
function register(string calldata name, address nameOwner, bytes32 secret, uint duration);
```

> Registers a .rsk name in RNS.

Reviewer: ```suggestion
Registers a `.rsk` name in RNS.
```

This method must be called after committing.

> Previously execute [`approve()`](https://github.com/riflabs/RIF-Token/blob/master/contracts/third-party/openzeppelin/token/ERC20/StandardToken.sol#L53) for the amount of tokens to be transferred.

Reviewer: ```suggestion
You must have previously executed [`approve()`](https://github.com/riflabs/RIF-Token/blob/master/contracts/third-party/openzeppelin/token/ERC20/StandardToken.sol#L53) for the amount of tokens to be transferred.
```

- `name` The name to register.
- `nameOwner` The owner of the name to register.
- `secret` The secret used to make the commitment.
- `duration` Time to register in years.

## Register via ERC-677

> Use RIF [`transferAndCall` method](https://github.com/riflabs/RIF-Token/blob/master/contracts/RIF/RIFToken.sol#L278) with the following encoding to perform a one-transaction RIF token transfer + domain registration.

Reviewer: ```suggestion
Use RIF [`transferAndCall` method](https://github.com/riflabs/RIF-Token/blob/master/contracts/RIF/RIFToken.sol#L278) with the following encoding to perform a RIF token transfer and domain registration in a single transaction
```
